packages/react-native-oursky/src/RequirePermission.js

```diff
@@ -0,0 +1,131 @@
+// @flow
+import * as React from "react";
+import { Platform } from "react-native";
+import Permissions from "react-native-permissions";
+import OpenSettings from "react-native-open-settings";
+
+import Dialog from "./Dialog";
+import type { LayoutProps as DialogLayoutProps } from "./Dialog";
+
+const AndroidPermission = ["photo"];
+const IOSPermission = ["notification", "photo"];
+
+export type Props = {
+  permission: "notification" | "photo",
+
+  UndeterminedDialogProps: DialogLayoutProps,
+  DeniedDialogProps: DialogLayoutProps,
+
+  onAccept: () => void,
+  onReject: () => void,
+};
+
+type Status = "authorized" | "denied" | "restricted" | "undetermined";
+
+type State = {
+  showUndeterminedDialog: boolean,
+  showDeniedDialog: boolean,
+};
+
+export default class RequirePermission extends React.PureComponent<
+  Props,
+  State
+> {
+  state = {
+    showUndeterminedDialog: false,
+    showDeniedDialog: false,
+  };
+
+  componentDidMount() {
+    const validPermission = Platform.select({
+      ios: IOSPermission,
+      android: AndroidPermission,
+    });
+    const requestPermission = this.props.permission;
+
+    if (validPermission.includes(requestPermission)) {
+      this.requestPermission(requestPermission);
+    }
+  }
+
+  requestPermission = async (permission: string) => {
```

Comment:
This method is named `requestPermission` however what it does is check permission. Can you change the name of this method?